app/controllers/entry_controller.rb

```diff
@@ -44,33 +42,25 @@ def create
   end
 
   def destroy
-    challenge = Challenge.find(params[:challenge])
-    entry = challenge.entries.find(params[:entry])
-
-    if challenge.owner?(current_user) || entry.owner?(current_user)
-      entry.destroy
-      challenge.save
-
-      flash[:notice] = "Deleted entry"
+    if @entry && (@challenge.owner?(current_user) || @entry.owner?(current_user))
```

Comment:
if we add nil checks in the `if` branch we'll need to add an `else` branch that does the redirect or renders the json. the before action keeps things a little dry. since we're using `find` in the before action it'll raise and be caught by the rescue and do the redirect for us